### DISCUSSION TITLE
incorrect battery voltage level & battery percentage when motors are on

### DISCUSSION THREAD

**Discussion by rithwikjayanth233** [bitcraze/discussions#857] [Q&A]
Hi,
I am trying to get the battery percentage and voltage level from my crazyflie drone using

    ```
cflib.crtp.init_drivers()
    with SyncCrazyflie(URI, cf=Crazyflie(rw_cache='./cache')) as scf:
        scf.cf.param.add_update_callback(group='deck', name='bcLighthouse4',
                                         cb=param_deck_lighthouse)
        time.sleep(1)
        
        logconf = LogConfig(name='Position', period_in_ms=10)
        logconf.add_variable('stateEstimate.x', 'float')
        logconf.add_variable('stateEstimate.y', 'float')
        logconf.add_variable('stateEstimate.z', 'float')
        logconf.add_variable('pm.vbat', 'FP16')
        logconf.add_variable('pm.batteryLevel', 'FP16')
        logconf.add_variable('pm.state', 'float')
```
however, i get the incorrect values
when the drone is flying, i.e., the motors are running, the battery percentage is always at 10. when the drone is stationary, the battery level is at 60-70. it doesnt show any other value like 12, 38 etc. only multiples of 10.

regarding the voltage level, when the drone is stationary, the voltage level is around 3.8 and when the drone is flying, it is at 3.4. 

Shouldnt the voltage level reduce with time? it just repeats the above cycle. 

Also a huge shoutout to the whole crazyflie community! I have gotten so many answers here. Y'all rock!

**Comment by tobbeanton** [comment]
Yeah, unfortunately the battery level is completely unreliable. Possibly a model could be made to improve it but because there is no current measurement done I don't think it will ever be good.
